src/components/pickers/BatchNumberPicker.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import classNames from 'classnames';
+import { useIntl } from 'react-intl';
+import { AsyncPaginate, LoadOptions } from 'react-select-async-paginate';
+
+import { itemDetailUrl } from 'stores/sagas/itemDetailSaga';
+import refreshAccessToken from 'stores/sagas/utils';
+import { getRequest } from 'utils/apiClient';
+
+import { BatchNumberPickerProp } from './types';
+
+const customStyles = {
+  control: (provided: any) => ({
+    ...provided,
+    background: '#fff',
+    border: 'none',
+    minHeight: '34px',
+    height: '34px',
+    boxShadow: 'none',
+    '&:hover': {
+      borderColor: '#B8BBBF',
+    },
+    fontSize: '14px',
+    overflow: 'hidden',
+  }),
+  placeholder: (provided: any) => ({
+    ...provided,
+    color: '#9ca3af',
+    fontSize: 14,
+  }),
+};
+
+function BatchNumberPicker({
+  field, watch, onChangeBatchNumber,
+}: BatchNumberPickerProp) {
+  const { formatMessage } = useIntl();
+  const format = useCallback((id: string, values: any = '') => formatMessage({ id }, values), [formatMessage]);
+
+  const [value, onChange] = useState(field);
+  const [isInvalid, setIsInvalid] = useState<boolean>(false);
+
+  const loadOptions: LoadOptions<any, any, { page: any }> = async (
+    searchQuery: any,
+    loadedOptions: any,
+    { page }: any,
+  ) => {
+    let response: any = { results: [] };
+    let isNext: boolean = false;
+
+    if (field !== undefined) {
+      await refreshAccessToken();
+      response = await getRequest(itemDetailUrl, { page, searcher: `${field?.lineItem.product_name} ${searchQuery}` }, true);
+      isNext = response.next !== null;
+    }
+    return {
+      options: response.results,
+      hasMore: isNext,
+      additional: {
+        page: page + 1,
+      },
+    };
+  };
+
+  useEffect(() => {
+    onChange(field);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [watch]);
+
+  useEffect(() => {
+    onChangeBatchNumber(value, field?.lineItem);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]);
+
+  useEffect(() => {
+    if (field?.batch_number === '' || field?.batch_number === null || field?.lineItem.batch_number === '' || field?.lineItem.batch_number === null || value === null) {
```

Comment:
Can we simplify this  'if' statement? Or is this impossible in that context? I know that there are a lot of 'dependencies' but I wonder if this line could be simplified. If not try do separate function for this check.

Reply:
To simplify some checks for such checks with 'null', '' or undefined 'lodash' could be good. You can take a look if you have time. Also here there is a good clue about such kind of check https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in.

Reply:
but as I wrote you can move this to separate function, It is just only proposal to improve this line.

Reply:
Here I can remove some statements, but I cannot use here your solution, because `batch_number` can be `undefined`, the same `value`.
I will change it to :
` if (field?.lineItem.batch_number === '' || field?.lineItem.batch_number === null || value === null) {`